browser/main/modals/PreferencesModal/UiTab.js

```diff
@@ -12,12 +13,27 @@ class UiTab extends React.Component {
     super(props)
 
     this.state = {
-      config: props.config
+      config: props.config,
+      options: {
+        lineNumbers: true,
+        theme: props.config.editor.theme,
+        readOnly: true,
+        mode: 'javascript'
```

Comment:
I think `lineNumbers`, `readOnly`, `mode` won't be changed.
You don't need to keep them in `state`.